app/src/main/java/com/example/myplaylistmaker/SearchActivity.kt

```diff
@@ -4,20 +4,40 @@ import android.os.Bundle
 import android.widget.EditText
 import android.widget.ImageView
 import android.content.Context
+import android.content.res.Configuration
 import android.text.Editable
 import android.text.TextWatcher
 import android.view.View
+import android.view.inputmethod.EditorInfo
 import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
 class SearchActivity : AppCompatActivity() {
 
     private lateinit var editText: EditText
     private lateinit var clearButton: ImageView
-    private var savedText: String? = null
     private lateinit var recyclerView: RecyclerView
     private lateinit var trackAdapter: TrackAdapter
+    private lateinit var apiService: ApiService
+    private lateinit var placeholderText: TextView
+    private lateinit var placeholderImage: ImageView
+    private lateinit var refreshButton: Button
+    private var trackList = arrayListOf<Track>()
```

Comment:
```suggestion
    private val trackList = arrayListOf<Track>()
```
Сама ссылка не меняется, только содержимое списка, поэтому можно сделать private val